Doc how kubelet client certificate renew v5 (#912)
SUSE/doc-caasp: adoc/admin-security-certificates.adoc

when the node is upgraded, all the kubeadm managed certificates and kubeconfigs, and kubelet client certificates get rotated.

. *Via the `kucero` addon*:

> if the administrator is not willing to upgrade the cluster, the `kucero` (KUbernetes control plane CErtificate ROtation) addon rotates all the kubeadm managed certificates and kubeconfigs. The kucero is a kubeadm checker/renewer in the form of a DaemonSet, which is deployed on master nodes. It's job is to periodically check and renew control plane kubeadm managed certificates/kubeconfigs.

Reviewer: ```suggestion
if the administrator is not willing to upgrade the cluster, the `kucero` (KUbernetes control plane CErtificate ROtation) addon rotates all the kubeadm managed certificates and kubeconfigs. The kucero is a kubeadm checker/renewer in the form of a DaemonSet, which is deployed on control plane nodes. It's job is to periodically check and renew control plane kubeadm managed certificates/kubeconfigs.
```

  .. A kubeadm checker/renewer in the form of a DaemonSet, which is deployed on master nodes. It's job is to periodically check and renew control plane kubeadm managed certificates/kubeconfigs.
  .. A kubelet server CSR signer controller, which signs the kubelet server CSR with a kubelet CA cert/key pair. The signed server certificates are stored in `/var/lib/kubelet/pki` and `/var/lib/kubelet/pki/kubelet-server-current.pem` is a symlink to the last generated server certificate.

.kubelet client/server certificate
[NOTE]
====
During the kubelet client/server certificate signing flow, kubelet sends the kubelet client/server CSR into the {kube} cluster. The `kube-controller-manager` signs the kubelet client CSR with the Kubernetes CA cert/key pair. The `kucero` signs the kubelet server CSR with the kubelet CA cert/key pair.

The kubelet client/server certificate renews automatically at approximately 70%-90% of the total lifetime of the certificate, the kubelet would use new client/server certificates without downtime.

> The kubelet client certificate uses the `/var/lib/kubelet/pki/kubelet-client-current.pem`, this file is a symlink to the latest signed client certificate.

Reviewer 2: ```suggestion
The kubelet client certificate is referred to `/var/lib/kubelet/pki/kubelet-client-current.pem`, this file is a symlink to the latest signed client certificate.
```

> The kubelet server certificate uses the `/var/lib/kubelet/pki/kubelet-server-current.pem`, this file is a symlink to the latest signed server certificate.

Reviewer 2: ```suggestion
The kubelet server certificate is referred to `/var/lib/kubelet/pki/kubelet-server-current.pem`, this file is a symlink to the latest signed server certificate.
```

commands sequentially on all master nodes.
====

=== Renewing Control Plane Certificates

* Replace kubeadm-managed certificates:

> . To SSH into the master node, renew all `kubeadm` certificates and reboot,

Reviewer: ```suggestion
. To SSH into the control plane node, renew all `kubeadm` certificates and reboot,
```

sudo cat /etc/kubernetes/admin.conf
----

* Replace the `kubelet` server certificate:
+
[IMPORTANT]
====
You need to generate `kubelet` server certificate for all the nodes on one of control plane nodes.
The `kubelet` CA certificate key only exists on the control plane nodes.

> Therefore, after generating re-signed `kubelet` server certificate/key for worker nodes, you have to copy each `kubelet` server certificate/key from the control plane node to the corresponding worker node.

Reviewer 2: ```suggestion
Therefore, after generating re-signed `kubelet` server certificate/key for worker nodes, you have to copy each `kubelet` server certificate/key from the control plane node to the corresponding worker nodes.
```

====
. Backup the original `kubelet` certificates and keys.
+
[source,bash]
----
sudo cp -r /var/lib/kubelet/pki /var/lib/kubelet/pki.bak
----
+

> . Sign each node `kubelet` server certificate with the CA certificate/key `/var/lib/kubelet/pki/kubelet-ca.crt` and `/var/lib/kubelet/pki/kubelet-ca.key`, make sure that the signed server certificate SAN is the same as the origin.

Reviewer 2: ```suggestion
. To sign each node `kubelet` server certificate with the CA certificate/key `/var/lib/kubelet/pki/kubelet-ca.crt` and `/var/lib/kubelet/pki/kubelet-ca.key`, make sure that the signed server certificate SAN is the same as the origin.
```
